include/ck/tensor_operation/gpu/device/impl/device_grouped_gemm_xdl_fixed_nk.hpp

```diff
@@ -0,0 +1,1014 @@
+                }
+
+                const auto K = arg.gemm_desc_kernel_arg_[i].a_grid_desc_ak0_m_ak1_.GetLength(I0) *
+                               arg.gemm_desc_kernel_arg_[i].a_grid_desc_ak0_m_ak1_.GetLength(I2);
+
+                if(GridwiseGemm::CalculateHasMainKBlockLoop(K) != has_main_k_block_loop)
+                {
+                    throw std::runtime_error("wrong! not all gemm has_main_k_block_loop");
+                }
+
+#if 1
+                grouped_gemm_kernel_args.push_back(
+                    GroupedGemmKernelArgument<NumDTensor>{arg.gemm_desc_kernel_arg_[i].a_ptr_,
+                                                          arg.gemm_desc_kernel_arg_[i].b_ptr_,
+                                                          arg.gemm_desc_kernel_arg_[i].ds_ptr_,
+                                                          arg.gemm_desc_kernel_arg_[i].e_ptr_,
+                                                          arg.gemm_desc_kernel_arg_[i].M_,
+                                                          arg.gemm_desc_kernel_arg_[i].N_,
+                                                          arg.gemm_desc_kernel_arg_[i].K_,
+                                                          arg.gemm_desc_kernel_arg_[i].StrideA_,
+                                                          arg.gemm_desc_kernel_arg_[i].StrideB_,
+                                                          arg.gemm_desc_kernel_arg_[i].StrideDs_,
+                                                          arg.gemm_desc_kernel_arg_[i].StrideE_});
+#endif
+            }
+
+            float ave_time = 0;
+
+            auto launch_kernel = [&](auto has_main_k_block_loop_) {
+                const auto kernel =
+                    kernel_grouped_gemm_xdl_fixed_nk<GridwiseGemm,
+                                                     GroupedGemmKernelArgument<NumDTensor>,
+                                                     GemmSpec,
+                                                     ALayout,
+                                                     BLayout,
+                                                     DsLayout,
+                                                     ELayout,
+                                                     DsDataType,
+                                                     Block2ETileMap,
+                                                     GroupedGemmBlock2ETileMap,
+                                                     AElementwiseOperation,
+                                                     BElementwiseOperation,
+                                                     CDEElementwiseOperation,
+                                                     has_main_k_block_loop_>;
+
+                const index_t grid_size_grp = arg.gemm_desc_kernel_arg_[0].BlockEnd_ -
+                                              arg.gemm_desc_kernel_arg_[0].BlockStart_;
```

Comment:
Since each gemm group launch same number of workgroups you can just divide `arg.grid_size_ / arg.group_count_`